crates/provider/src/provider/eth_call/params.rs

```diff
@@ -52,18 +55,40 @@ where
     pub const fn block(&self) -> Option<BlockId> {
         self.block
     }
+
+    /// Sets the block overrides for this call.
+    pub fn with_block_overrides(mut self, overrides: BlockOverrides) -> Self {
+        self.block_overrides = Some(overrides);
+        self
+    }
+
+    /// Returns a reference to the block overrides if set.
+    pub const fn block_overrides(&self) -> Option<&BlockOverrides> {
+        self.block_overrides.as_ref()
+    }
 }
 
 impl<N: Network> serde::Serialize for EthCallParams<N> {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let len = if self.overrides().is_some() { 3 } else { 2 };
+        let len = match (self.block_overrides(), self.overrides()) {
+            (Some(_), Some(_)) => 4,
+            (Some(_), None) => 3,
+            (None, Some(_)) => 3,
+            (None, None) => 2,
+        };
 
         let mut seq = serializer.serialize_seq(Some(len))?;
         seq.serialize_element(&self.data())?;
 
-        if let Some(overrides) = self.overrides() {
+        if let Some(block_overrides) = self.block_overrides() {
+            seq.serialize_element(block_overrides)?;
             seq.serialize_element(&self.block().unwrap_or_default())?;
-            seq.serialize_element(overrides)?;
+            if let Some(state_overrides) = self.overrides() {
```

Comment:
here we need to unwrap_or_default because blockoverrides must go last

Reply:
op catch